go/private/mode.bzl

```diff
@@ -93,6 +93,7 @@ def get_mode(ctx, go_toolchain, cgo_context_info, go_config_info):
     goos = go_toolchain.default_goos if getattr(ctx.attr, "goos", "auto") == "auto" else ctx.attr.goos
     goarch = go_toolchain.default_goarch if getattr(ctx.attr, "goarch", "auto") == "auto" else ctx.attr.goarch
     gc_goopts = go_config_info.gc_goopts if go_config_info else []
+    pgoprofile = go_config_info.pgoprofile.files.to_list()[0] if go_config_info and len(go_config_info.pgoprofile.files.to_list()) > 0 else None
```

Comment:
Let's fail here if the length is `>1`, otherwise making changes to that case (such as merging) in the future would be a breaking change. Better to be restrictive now.

Reply:
sgtm